ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveFilterJoinRule.java

```diff
@@ -130,7 +130,7 @@ public void onMatch(RelOptRuleCall call) {
 
   public static class HiveFilterJoinTransposeRule extends HiveFilterJoinRule {
     public HiveFilterJoinTransposeRule() {
-      super(RelOptRule.operand(Join.class, RelOptRule.any()), "HiveFilterJoinRule:no-filter", true,
+      super(RelOptRule.operand(Join.class, RelOptRule.any()), "HiveFilterJoinTransposeRule", true,
```

Comment:
this could also be handled by `getClass().getSimpleName()` - I think that would have worked in the superclass as well

Reply:
Thanks @kgyrtkirk, that's better so we don't risk getting out of sync in case the class name gets changed. I have updated the PR.